### ISSUE TITLE
Invalid plan for DISTINCT on user-defined datatype,

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#4529]
ORCA generates an invalid plan, for this query with 'hstore':

```
-- Install 'hstore':
\i hstore.sql

postgres=# CREATE TABLE testhstore (h hstore);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause, and no column type is suitable for a distribution key. Creating a NULL policy entry.
CREATE TABLE
postgres=# insert into testhstore values ('');
INSERT 0 1
postgres=# set optimizer=on;
SET
postgres=# select count(distinct h) from testhstore ;
ERROR:  Type 16837 is not hashable.  (seg0 slice1 127.0.0.1:40000 pid=29901)
```

**Comment by hlinnaka** [comment]
This was exposed by new regression tests that were just added in the 9.0 merge, along with new btree and hash operator classes for hstore. Even though this is a new issue with hstore, as it didn't have those operator classes before, in principle this is a pre-existing bug that could happen with any user-defined datatype. I think for this to happen, the datatype must have b-tree and hash opclasses, and it must not be "GPDB hashable", i.e. not a built-in type.

**Comment by hlinnaka** [comment]
Here's the plan:
```
postgres=# explain select count(distinct h) from testhstore ;
                                                 QUERY PLAN                                                 
------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=0.00..431.00 rows=1 width=8)
   ->  Gather Motion 3:1  (slice2; segments: 3)  (cost=0.00..431.00 rows=1 width=5)
         ->  GroupAggregate  (cost=0.00..431.00 rows=1 width=5)
               Group Key: h
               ->  Sort  (cost=0.00..431.00 rows=1 width=5)
                     Sort Key: h
                     ->  Redistribute Motion 3:3  (slice1; segments: 3)  (cost=0.00..431.00 rows=1 width=5)
                           Hash Key: h
                           ->  GroupAggregate  (cost=0.00..431.00 rows=1 width=5)
                                 Group Key: h
                                 ->  Sort  (cost=0.00..431.00 rows=1 width=5)
                                       Sort Key: h
                                       ->  Table Scan on testhstore  (cost=0.00..431.00 rows=1 width=5)
 Optimizer: PQO version 2.54.2
(14 rows)
```
Should not create a Motion with a non-gpdb-hashable datatype in the Merge Key.

**Comment by vraghavan78** [comment]
Thanks for reporting. Will take a look very soon.

**Comment by khannaekta** [comment]
Issue fixed in GPORCA v2.55.10 (commit: https://github.com/greenplum-db/gporca/commit/b31aa174f6caa88f75c0b037e52ebf717e21bfd7)
